pkg/querier/blocks_store_queryable.go

```diff
@@ -136,7 +140,51 @@ func newBlocksStoreQueryableMetrics(reg prometheus.Registerer) *blocksStoreQuery
 			Name: "cortex_querier_query_storegateway_chunks_total",
 			Help: "Number of chunks received from store gateways at query time.",
 		}),
+
+		concurrencyWait: promauto.With(reg).NewHistogram(prometheus.HistogramOpts{
+			Name:                            "cortex_querier_storegateway_concurrency_wait_seconds",
+			Help:                            "Time from query start until the last store-gateway started processing the query.",
```

Comment:
Is this more accurate? "query start" seems ambiguous to me, and "last store-gateway" isn't quite correct:

```suggestion
			Help:                            "Longest wait time amongst all store-gateways that serviced a request, where wait time is the time between the request arriving at that store-gateway and when it began processing the request.",
```